crates/widgets/src/tab_widget.rs

```diff
@@ -0,0 +1,382 @@
+
+        /// Sets or shares the padding property.
+        padding: Thickness,
+
+        /// Sets or shares the foreground property.
+        foreground: Brush,
+
+        /// Sets or shares the text property.
+        text: String16,
+
+        /// Sets or share the font size property.
+        font_size: f64,
+
+        /// Sets or shares the font property.
+        font: String,
+
+        /// Sets or shares the icon property.
+        icon: String,
+
+        /// Sets or shares the icon brush property.
+        icon_brush: Brush,
+
+        /// Sets or share the icon font size property.
+        icon_size: f64,
+
+        /// Sets or shares the icon font property.
+        icon_font: String,
+
+        /// Sets or shares the selected property.
+        selected: bool,
+
+        /// Sets or shares the spacing between icon and text.
+        spacing: f64,
+
+        /// Sets or shares the spacing between icon and text.
+        close_button: bool
+    }
+);
+
+impl TabHeader
+{
+    pub fn on_header_click<T: 'static + Fn(&mut StatesContext, Point) -> bool>(mut self,callback: T)->Self
+    {
+        self.state.on_header_click_callback = Some(Box::new(callback));
+        self
+    }
+    pub fn on_close_click<T: 'static + Fn(&mut StatesContext, Point) -> bool>(mut self,callback: T)->Self
+    {
+        self.state.on_close_click_callback = Some(Box::new(callback));
+        self
+    }
+}
+
+impl Template for TabHeader {
+    fn template(mut self, id: Entity, ctx: &mut BuildContext) -> Self {
+        let mut toggle_button = ToggleButton::new()
+        .background(id)
+        .border_radius(id)
+        .border_width(id)
+        .border_brush(id)
+        .padding(id)
+        .foreground(id)
+        .text(id)
+        .font_size(id)
+        .font(id)
+        .icon(id)
+        .icon_brush(id)
+        .icon_size(id)
+        .icon_font(id)
+        .selected(id)
+        .spacing(id);
+
+        if let Some(callback) = self.state.on_header_click_callback.take() {toggle_button = toggle_button.on_click(callback);}
+
+        let mut button = Button::new()
+        .background(id)
+        .border_radius(id)
+        .border_width(id)
+        .border_brush(id)
+        .padding(id)
+        .foreground(id)
+        .font_size(id)
+        .font(id)
+        .icon(id)
+        .icon_brush(id)
+        .icon_size(id)
+        .icon_font(id)
+        .spacing(id)
+        .pressed(false)
+        .text("X")
+        .width(2.0);
+
+        if let Some(callback) = self.state.on_close_click_callback.take() {button = button.on_click(callback);}
+        //if self.close_button() == false {button = button.visibility(Visibility::Collapsed);}
+
+        self.name("TabWidget")
+        .selected(false)
+        .height(36.0)
+        .min_width(64.0)
+        .background(colors::LYNCH_COLOR)
+        .border_radius(4.0)
+        .border_width(0.0)
+        .border_brush("transparent")
+        .padding((16.0, 0.0, 16.0, 0.0))
```

Comment:
Could you do me a favor and remove all .0s? e.g. `.padding((16, 0, 16, 0))`

Reply:
I was sure that they all require floats, sorry

Reply:
not any more. No problem